src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+
+import logout from '@/apis/logout';
+import { ILogin, ILoginState } from '@/pages/Account/Login/types';
+
+const useAuth = () => {
+  const navigate = useNavigate();
+  const [loginState, setLoginState] = useState<ILoginState>(() => {
+    const accessToken = sessionStorage.getItem('accessToken') || null;
+    const refreshToken = sessionStorage.getItem('refreshToken') || null;
+    const expires = parseInt(sessionStorage.getItem('expires') || '0', 10) || null;
+
+    return { accessToken, expires, refreshToken };
+  });
+
+  const handleLoginResponse = (response: ILogin) => {
```

Comment:
TS return type is missing. Please make sure that you update ESLint rules, so explicitly adding return types is required.